src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const checkSep = (separator === null) ? 'null' : separator;
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
```

Comment:
not totally sure, but do you need an explicit check for `null` and `undefined` or could just go with a generic check `if (this[i])` that would result in false if this[i] is any equivalent of a `falsy` value

Reply:
yeah for first i wanted to do that way, but in this case its gonna ignore all false values like 0 or NaN, and it sould not to be ignored by the code.